src/main/java/net/ttddyy/dsproxy/proxy/jdk/JdkJdbcProxyFactory.java

```diff
@@ -75,15 +72,29 @@ public CallableStatement createCallableStatement(CallableStatement callableState
      */
     @Override
     public ResultSet createResultSet(ResultSet resultSet, ConnectionInfo connectionInfo, ProxyConfig proxyConfig) {
+        ResultSetProxyLogicFactory factory = proxyConfig.getResultSetProxyLogicFactory();
         // when proxy logic factory is specified, create a proxy
-        if (proxyConfig.getResultSetProxyLogicFactory() != null) {
+        if (factory != null) {
             return (ResultSet) Proxy.newProxyInstance(ProxyJdbcObject.class.getClassLoader(),
                     new Class[]{ProxyJdbcObject.class, ResultSet.class},
-                    new ResultSetInvocationHandler(resultSet, connectionInfo, proxyConfig));
+                    new ResultSetInvocationHandler(factory, resultSet, connectionInfo, proxyConfig));
         } else {
             return resultSet;
         }
     }
 
+    @Override
+    public ResultSet createGeneratedKeys(Statement statement, ConnectionInfo connectionInfo, ProxyConfig proxyConfig) throws SQLException {
+        ResultSetProxyLogicFactory factory = proxyConfig.getGeneratedKeysProxyLogicFactory();
+        // when proxy logic factory is specified, create a proxy
+        if (factory != null) {
+            return (ResultSet) Proxy.newProxyInstance(ProxyJdbcObject.class.getClassLoader(),
+                    new Class[]{ProxyJdbcObject.class, ResultSet.class},
+                    new ResultSetInvocationHandler(factory, statement.getGeneratedKeys(), connectionInfo, proxyConfig));
```

Comment:
Closely looking the change, I think the `ResultSet` returned by this `statement.getGeneratedKeys()` need to be closed.
This makes things bit more complicated.

What I think is:
- In `~ProxyLogic` always create repeatable read proxy, and consume it and close it there. Then, use the already consumed `ResultSet` in `ExecutionInfo#getGeneratedKeys()` and in application code where it calls `getGeneratedKeys()`

What do you think?

Reply:
I was thinking more.
Another way is, create a `GeneratedKeys` class (internally use `Map<String, List<Object>>` to hold the `ResultSet`) and used it for `ExecutionInfo`. 

In `~ProxyLogic` class,
- create a ResultSet proxy based on user specified factory
- read the `ResultSet` and create `GeneratedKeys`
- close the `ResultSet`
- make the proxy gets returned if `getGeneratedKeys` method is called

So, when generated-key-proxy-factory is specified, it always read and close the `ResultSet`. Then, `ExecutionInfo#getGeneratedKeys()` returns `GeneratedKeys` object.
If repeatable resultset is specified, user app can still call `getGeneratedKeys`. If simple proxy factory is specified, when user call `getGeneratedKeys()` result set is already closed.
(For userbility perspective, it may be good to always return repeated read proxy...)


I think key points are:
- use `ResultSet` or `GeneratedKeys` object for `ExecutionInfo#getGeneratedKeys()`
- Always use repeatable read proxy or user specified strategy

thought?

Reply:
I think it is a little overkill to create wrapper like `GeneratedKeys`. Also `GeneratedKeys` would break the `homogeneity` of what I saw of the project.
So for the first point, I prefer to use a simple ResultSet.

For the second point, I prefere to let user specify a strategy like it seems to be done in the rest of the system.

For your concerns about the fact that the `ResultSet` may not be closed, is it really an issue? since:
- generated keys `ResulSet` is opened while its `Statement` is already opened
- generated keys `ResulSet` will be automatically closed, when its `Statement` closes

Reply:
Ok, I'll merge the current PR as is and will add a bit of modification but the main part will be what's in this PR.
About closing `ResultSet`, I think it's better to close it as a framework being well behaved even though it will automatically be closed. I'll add `ResultSet#isClosed()` check, then I just found a bug for that in repeatable read proxy.

Anyway, greatly appreciated this generated keys related changes!!